### ISSUE TITLE
SUPPA cluster events error

### ISSUE THREAD

**Issue by paolo-kunderfranco** [nf-core/rnasplice#71]
### Description of the bug

Hi,
I am trying dev version and I encountered the following error:

I have 32 samples total, 4 conditions, 8 samples per condition

Best
```

Error executing process > 'NFCORE_RNASPLICE:RNASPLICE:SUPPA_SALMON:CLUSTEREVENTS_IOE (CP-Cortex)

Caused by:
  Process `NFCORE_RNASPLICE:RNASPLICE:SUPPA_SALMON:CLUSTEREVENTS_IOE (CP-Cortex)` terminated with an error exit status (1)

Command executed:

  suppa.py \
      clusterEvents \
      --dpsi CP-Cortex_local_diffsplice.dpsi \
      --psivec CP-Cortex_local_diffsplice.psivec \
      --dpsi-threshold 0.05 \
      --eps 0.05 \
      --metric euclidean \
      --min-pts 20 \
      --groups 1-25,6-27,12-30,17-32 \
      --clustering DBSCAN \
        -o CP-Cortex_local_cluster
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_RNASPLICE:RNASPLICE:SUPPA_SALMON:CLUSTEREVENTS_IOE":
      suppa: $(python -c "import pkg_resources; print(pkg_resources.get_distribution('suppa').version)")
  END_VERSIONS

Command exit status:
  1

Command output:
  (empty)

Command error:
  ERROR:lib.cluster_tools:Invalid index. Index 32 is larger than the number of columns in the file (16).`


```
### Command used and terminal output

```console
nextflow run nf-core/rnasplice --input $wd/samplesheet.csv \
							   --contrasts $wd/contrast.csv \
							   --outdir splice_attempt1 \
							   --genome GRCm38 \
							   --aligner star_salmon \
							   --min_samps_gene_expr 2 \
							   --min_samps_feature_expr 2 \
							   --min_gene_expr 5 \
							   --min_feature_expr 5 \
							   --min_feature_prop 0.2 \
							   --dexseq_exon true \
							   --edger_exon true\
							   --dexseq_dtu true \
							   --dtu_txi scaledTPM \
							   --rmats true \
							   --rmats_read_len 75 \
							   --suppa true \
							   --suppa_per_isoform true \
							   --save_reference true \
							   --sashimi_plot true \
							   -config config.config \
							   -profile singularity \
							   -r dev
```

### Relevant files

[nextflow.log](https://github.com/nf-core/rnasplice/files/11890872/nextflow.log)

### System information

nextflow/22.10.4
Debian 10 
Slurn

**Comment by expansion-bioemformatics** [comment]
I am having the same issue running via Tower on AWS Batch.

`Workflow execution completed unsuccessfully
The exit status of the task that caused the workflow execution to fail was: 1

Error executing process > 'NFCORE_RNASPLICE:RNASPLICE:SUPPA_SALMON:CLUSTEREVENTS_IOE (ALS1_DMSO-WT4_DMSO)'

Caused by:
  Essential container in task exited

Command executed:

  suppa.py \
      clusterEvents \
      --dpsi ALS1_DMSO-WT4_DMSO_local_diffsplice.dpsi \
      --psivec ALS1_DMSO-WT4_DMSO_local_diffsplice.psivec \
      --dpsi-threshold 0.05 \
      --eps 0.05 \
      --metric euclidean \
      --min-pts 20 \
      --groups 1-3,4-6,7-9,10-12,13-15,16-18,19-21,22-24,25-27,28-30,31-33,34-36,37-39,40-42,43-45,46-48 \
      --clustering DBSCAN \
        -o ALS1_DMSO-WT4_DMSO_local_cluster
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_RNASPLICE:RNASPLICE:SUPPA_SALMON:CLUSTEREVENTS_IOE":
      suppa: $(python -c "import pkg_resources; print(pkg_resources.get_distribution('suppa').version)")
  END_VERSIONS

Command exit status:
  1

Command output:
  (empty)

Command error:
  ERROR:lib.cluster_tools:Invalid index. Index 48 is larger than the number of columns in the file (6).
  [90m6:51PM[0m [32mINF[0m shutdown filesystem start
  [90m6:51PM[0m [32mINF[0m shutdown filesystem done`

**Comment by asmaali98** [comment]
Hi @paolo-kunderfranco and @expansion-bioemformatics , thanks for raising the issue. We will work on having this error resolved.

**Comment by dkoppstein** [comment]
I am also getting a similar issue. I am not sure, but I think the problem occurs because all contrasts are passed to suppa via `--groups`, whereas the PSI files are somehow already specific for the contrast of interest. Therefore, there are more columns than samples.

**Comment by dkoppstein** [comment]
To be clear, this happens whenever there are >2 contrasts specified in the samplesheet. I was able to "solve" the problem by running the pipeline multiple times, once for each desired contrast. However, there should be a contrastsheet with multiple contrasts specified in one of the test configs.

**Comment by amizeranschi** [comment]
Hi @dkoppstein 

What do you mean by `whenever there are >2 contrasts specified in the samplesheet` ? 

Did you mean `>2 contrasts in the contrast sheet`? Or `>2 samples in the samplesheet`?

I'm also running into this issue, with the following samplesheet and contrast sheet:

```
sample,condition,strandedness,fastq_1,fastq_2
K562_TARDBP_TRT_REP1,K562_TARDBP_TRT,reverse,fastq-test/SRR3469414-R1.fastq.gz,fastq-test/SRR3469414-R2.fastq.gz
K562_TARDBP_TRT_REP2,K562_TARDBP_TRT,reverse,fastq-test/SRR3469415-R1.fastq.gz,fastq-test/SRR3469415-R2.fastq.gz
K562_TARDBP_CTL_REP1,K562_TARDBP_CTL,reverse,fastq-test/SRR3469418-R1.fastq.gz,fastq-test/SRR3469418-R2.fastq.gz
K562_TARDBP_CTL_REP2,K562_TARDBP_CTL,reverse,fastq-test/SRR3469419-R1.fastq.gz,fastq-test/SRR3469419-R2.fastq.gz
HepG2_TARDBP_TRT_REP1,HepG2_TARDBP_TRT,reverse,fastq-test/SRR4421909-R1.fastq.gz,fastq-test/SRR4421909-R2.fastq.gz
HepG2_TARDBP_TRT_REP2,HepG2_TARDBP_TRT,reverse,fastq-test/SRR4421908-R1.fastq.gz,fastq-test/SRR4421908-R2.fastq.gz
HepG2_TARDBP_CTL_REP1,HepG2_TARDBP_CTL,reverse,fastq-test/SRR4421557-R1.fastq.gz,fastq-test/SRR4421557-R2.fastq.gz
HepG2_TARDBP_CTL_REP2,HepG2_TARDBP_CTL,reverse,fastq-test/SRR4421558-R1.fastq.gz,fastq-test/SRR4421558-R2.fastq.gz
```

```
contrast,treatment,control
K562_TARDBP_TRT_CTL,K562_TARDBP_TRT,K562_TARDBP_CTL
HepG2_TARDBP_TRT_CTL,HepG2_TARDBP_TRT,HepG2_TARDBP_CTL
```

**Comment by amizeranschi** [comment]
Actually, my error is very similar, but not identical to the one above:

```
ERROR ~ Error executing process > 'NFCORE_RNASPLICE:RNASPLICE:SUPPA_SALMON:CLUSTEREVENTS_IOI (HepG2_TARDBP_TRT-HepG2_TARDBP_CTL)'

Caused by:
  Process `NFCORE_RNASPLICE:RNASPLICE:SUPPA_SALMON:CLUSTEREVENTS_IOI (HepG2_TARDBP_TRT-HepG2_TARDBP_CTL)` terminated with an error exit status (1)

Command executed:

  suppa.py \
      clusterEvents \
      --dpsi HepG2_TARDBP_TRT-HepG2_TARDBP_CTL_transcript_diffsplice.dpsi \
      --psivec HepG2_TARDBP_TRT-HepG2_TARDBP_CTL_transcript_diffsplice.psivec \
      --dpsi-threshold 0.05 \
      --eps 0.05 \
      --metric euclidean \
      --min-pts 20 \
      --groups 1-2,3-4,5-6,7-8 \
      --clustering DBSCAN \
        -o HepG2_TARDBP_TRT-HepG2_TARDBP_CTL_transcript_cluster
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_RNASPLICE:RNASPLICE:SUPPA_SALMON:CLUSTEREVENTS_IOI":
      suppa: $(python -c "import pkg_resources; print(pkg_resources.get_distribution('suppa').version)")
  END_VERSIONS

Command exit status:
  1

Command output:
  (empty)

Command error:
  ERROR:lib.cluster_tools:Invalid index. Index 8 is larger than the number of columns in the file (4).
```

**Comment by dkoppstein** [comment]
Sorry -- I did indeed mean `>2 contrasts in the contrast sheet`. And it is clear that the error is similar, i.e. that all groups are specified on the command line, even though the dpsi and psivec files are already restricted to the contrast specified.

**Comment by amizeranschi** [comment]
Good catch, I think you are right and the error seems to be caused by how the `--groups` setting is being specified by the pipeline.

**Comment by jma1991** [comment]
Thanks for the comments everyone. I'm working on pushing a fix for this today. Hopefully report back with a PR!

**Comment by paolo-kunderfranco** [comment]
exactly, I did have more contrast specified and the pipelne failed in SUPP.py

`  ERROR:lib.cluster_tools:Invalid index. Index 32 is larger than the number of columns in the file (16).`

I am trying now to run again with only one contrast specified.

**Comment by paolo-kunderfranco** [comment]
update, with only one contrast specified, CLUSTEREVENTS_IOE completed succesfully, however I am now facing and error 
with `NFCORE_RNASPLICE:RNASPLICE:DRIMSEQ_DEXSEQ_DTU_SALMON:STAGER`, 
I do need to open a new issue?
Best

**Comment by tijeco** [comment]
I'm not sure if this is helpful, but I've looked into this before and had a similar error as @paolo-kunderfranco   `ERROR:lib.cluster_tools:Invalid index. Index 32 is larger than the number of columns in the file (16).` I noticed this happened if the samplesheet wasn't sorted by condition, because it seems like suppa assumes that all the samples are sorted by condtion.

**Comment by jma1991** [comment]
Thanks for your patience on this bug. I believe we have fixed the issue with #86. Please can you pull from dev and see if this works.

**Comment by jma1991** [comment]
Closing because all tests are passing and fix has been merged into dev branch. Please re-open if you still encounter an error.
